ts/src/program/event.ts

```diff
@@ -49,7 +58,7 @@ export class EventManager {
    */
   private _onLogsSubscriptionId: number | undefined;
 
-  constructor(programId: PublicKey, provider: Provider, coder: Coder) {
+  constructor(programId: PublicKey, provider: Provider | undefined, coder: Coder) {
```

Comment:
Why is this now undefined?

Reply:
There was some recent change that forced this. Some upstream caller doesn't have provider anymore. I think it started at Program level